### PR TITLE
webauth-i-challenge

### PR DESCRIPTION
**Pull request by vijaydas**

### CODE REVIEW

**Comment by vijaydas** [comment]
day 1 MVP

File: api/server.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+server.use(cors());
+
+server.get('/', (req, res) => {
+  res.send("It's alive!");
+});
+
+server.use('/api/users', usersRouter);
+server.use('/api/auth', authRouter);
+
+module.exports = server;
```
Comment (angeltorgon) [inline api/server.js:21]: ✔️

File: auth/auth-router.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      } else {
+        res.status(401).json({ message: 'Invalid Credentials' });
+      }
+    })
+    .catch(error => {
+      res.status(500).json(error);
+    });
+});
+
+module.exports = router;
```
Comment (angeltorgon) [inline auth/auth-router.js:40]: ✔️

File: auth/restricted.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+          res.status(401).json({ message: 'Invalid Credentials' });
+        }
+      })
+      .catch(error => {
+        res.status(500).json(error);
+      });
+  } else {
+    res.status(400).json({ message: 'Please provide credentials' });
+  }
+};
```
Comment (angeltorgon) [inline auth/restricted.js:25]: ✔️

File: users/users-model.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+      const [id] = ids;
+      return findById(id);
+    });
+}
+
+function findById(id) {
+  return db('users')
+    .where({ id })
+    .first();
+}
```
Comment (angeltorgon) [inline users/users-model.js:31]: ✔️

File: users/users-router.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+router.get('/', restricted, (req, res) => {
+  Users.find()
+    .then(users => {
+      res.json(users);
+    })
+    .catch(err => res.send(err));
+});
+
+module.exports = router;
```
Comment (angeltorgon) [inline users/users-router.js:16]: ✔️

**Review by angeltorgon — COMMENTED** [review]
## Minimum Viable Product Objectives
* [x] POST register endpoint that creates a user and hashes password
* [x] POST login endpoint that authenticates user
* [x] GET endpoint that returns all users if user if logged in

## Stretch Objectives
* [x] Add middleware that insures a user is logged in when accessing routes

# Project Manager Feedback
## Great
* You completed MVP objectives! :heavy_check_mark:
* You completed stretch objectives! :heavy_check_mark:
Great work Vijay!
## Requested Improvements
* Create a react app that will user your api!
## Questions
* How comfortable do you feel authenticating a user?
## Rating {3}
:star: :star: :star:

**Pull request merged**